### PR TITLE
npm support

### PR DESCRIPTION
**Pull request by wsikorska-codilime**
Introduced motto which will allows us to integrate npm packages inside javascript extensions. Now we can use require function to integrate module installed via npm in node_modules at working directory. Migration from dop251/otto to robertkrimen/otto was necessary to make it work. robertkrimen otto seems to be more alive than dop251.

### CODE REVIEW

File: examples/npm_example/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Using NPM modules in extensions
+-------------------------------
+
+In this example, we show how we can
+use modules installed via npm in js extensions
+
+We have installed npm module underscore in
```
Comment (marcin-ptaszynski) [inline examples/npm_example/README.md]: Registering modules should also be documented.

File: examples/npm_example/example_schema.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    - status
+    required: []
+    type: object
+  singular: pet
+  title: Pet
+extensions:
+- id: schema
+  code_type: javascript
+  code: |
+    gohan_register_handler("pre_list", function (context){
```
Comment (marcin-ptaszynski) [inline examples/npm_example/example_schema.yaml]: Could you give an example of registering module too?

File: docs/source/extension.rst [hunk 3]
```diff
@@ -294,7 +294,8 @@ close ssh session
 
 - require(module)
 
-Dynamically load modules
+Dynamically load modules loaded in source code or
+installed via npm in node_modules at working directory
```
Comment (marcin-ptaszynski) [inline docs/source/extension.rst]: Is working directory where server is executed? I think it should be configurable path from config.

Comment (wsikorska-codilime) [inline docs/source/extension.rst]: Ok, I'll add a configurable path in config. Working directory is a directory of loaded config yaml.

**Comment by nati** [comment]
wow this is pretty neat.
pl rebase code

**Pull request merged**